icu4j/main/classes/core/src/com/ibm/icu/message2/PluralSelectorFactory.java

```diff
@@ -0,0 +1,117 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+import java.util.Locale;
+import java.util.Map;
+
+import com.ibm.icu.number.FormattedNumber;
+import com.ibm.icu.text.FormattedValue;
+import com.ibm.icu.text.PluralRules;
+import com.ibm.icu.text.PluralRules.PluralType;
+
+/**
+ * Creates a {@link Selector} doing plural selection, similar to <code>{exp, plural}</code>
+ * in {@link com.ibm.icu.text.MessageFormat}.
+ */
+class PluralSelectorFactory implements SelectorFactory {
+    private final PluralType pluralType;
+
+    /**
+     * Creates a {@code PluralSelectorFactory} of the desired type.
+     *
+     * @param type the kind of plural selection we want
+     */
+    PluralSelectorFactory(String type) {
```

Comment:
Please add a TODO to use an enum -- probably PluralType -- not a String type.

Reply:
Done.